shm/Game.cpp

```diff
@@ -9,4 +12,32 @@ bool Game::checkWinCondition() const {
 
 bool Game::checkLoseCondition() const {
     return (player_->getMoney() == 0);
-}
\ No newline at end of file
+}
+
+void Game::Travel() {
+    std::cout << *map_; //unimplemented << operator for map
+    auto destination = map_->getIsland(getTravelLocation());
+    map_->setCurrentPosition(destination);
+    auto distance = map_->getDistanceToIsland(destination);
+    advanceTimeTraveling(distance);
```

Comment:
Should `map_->setCurrentPosition(destination);` be placed before calculating the distance? I'd rather do:


    auto distance = map_->getDistanceToIsland(destination);
    advanceTimeTraveling(distance);
    map_->setCurrentPosition(destination);

Reply:
Why? It changes nothing. There is no functional reason for time advancement to go before location change.

Reply:
Only thing that needs changing is calculating distance before travelling :)

Reply:
For sure calculating distance should be done before travelling.

setting current position to destination for me is more logically correct when happens at the and of travelling process, so at the end of Travel method